cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/validation/DelegatingSeekableInputStream.java

```diff
@@ -0,0 +1,88 @@
+/*
+ *  Copyright © 2022 Cask Data, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ *  use this file except in compliance with the License. You may obtain a copy of
+ *  the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+package io.cdap.cdap.etl.api.validation;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * A helper for implementing a {@link SeekableInputStream} where subclasses only need to implement the
+ * methods not present in java's InputStream.
+ */
+public abstract class DelegatingSeekableInputStream extends SeekableInputStream {
+  private final InputStream inputStream;
+
+  public DelegatingSeekableInputStream(InputStream inputStream) {
+    this.inputStream = inputStream;
+  }
+
+  /**
+   * Seek to the given offset from the start of the file.
+   * The next read() will be from that location. Can't seek past the end of the file.
+   */
+  public abstract void seek(long pos) throws IOException;
```

Comment:
this and following method are already defined in `SeekableInputStream`, should we remove them here?

Reply:
ah yes, should remove, thanks